src/components/common/PageSearch/PageSearch.tsx

```diff
@@ -0,0 +1,79 @@
+import ArrowDownwardIcon from '@mui/icons-material/ArrowDownward'
+import ArrowUpwardIcon from '@mui/icons-material/ArrowUpward'
+import SearchIcon from '@mui/icons-material/Search'
+import { Box, IconButton, InputAdornment, TextField } from '@mui/material'
+import { observer } from 'mobx-react-lite'
+import React, { ChangeEvent, useEffect, useState } from 'react'
+
+import { useSharedStore } from '../../SharedStoreProvider'
+
+import { useStyles } from './PageSearch.styles'
+
+const PageSearchInput = () => {
+    const [inputValue, setInputValue] = useState('')
+    const { classes } = useStyles()
+    const {
+        documentStore: {
+            documentSearchStore: { isOpen, setQuery, activeSearch },
+        },
+    } = useSharedStore()
+
+    const handleInputChange = (event: ChangeEvent<HTMLInputElement>): void => {
+        setInputValue(event.target.value)
+    }
+
+    useEffect(() => {
+        setQuery(inputValue)
+    }, [inputValue, setQuery])
```

Comment:
and use reaction inside store to update query with debounce

Reply:
same as above

Reply:
`useEffect` hook do makes code harder to understand because it makes some calls based on some changes, and you can place it anywhere in any component
and when the number of that kind of hooks increase it makes almost impossible to understand what's happening